stdlib/public/Synchronization/Mutex/WasmImpl.swift

```diff
@@ -13,30 +13,34 @@
 // Note: All atomic accesses on WASM are sequentially consistent regardless of
 // what ordering we tell LLVM to use.
 
-@_extern(c, "llvm.wasm32.memory.atomic.wait32")
+@_extern(c, "llvm.wasm.memory.atomic.wait32")
 internal func _swift_stdlib_wait(
   on: UnsafePointer<UInt32>,
   expected: UInt32,
   timeout: Int64
 ) -> UInt32
 
-@_extern(c, "llvm.wasm32.memory.atomic.notify")
-internal func _swift_stdlib_wake(on: UnsafePointer<UInt32>, count: UInt32)
+@_extern(c, "llvm.wasm.memory.atomic.notify")
+internal func _swift_stdlib_wake(on: UnsafePointer<UInt32>, count: UInt32) -> UInt32
 
 extension Atomic where Value == _MutexHandle.State {
   internal borrowing func _wait(expected: _MutexHandle.State) {
-    _swift_stdlib_wait(
+    #if _runtime(_multithreaded)
```

Comment:
I think we should generally make `Mutex` unavailable when the runtime isn't multithreaded

Reply:
although I guess it doesn't really matter because `Mutex` can always acquire in the single threaded context and needing to call `wait` or `notify` should ever occur so this is probably fine.

Reply:
Ah, it might be better to have a separate noop implementations for single threaded targets instead of just skipping wait/notify operations.